src/ScillaContractDeployer.ts

```diff
@@ -152,14 +163,18 @@ async function deploy_from_file(
 
   const code = read(path);
   const contract = setup.zilliqa.contracts.new(code, init);
-  const [_, sc] = await contract.deploy(
+  const [tx, sc] = await contract.deploy(
     { ...setup },
     setup.attempts,
     setup.timeout,
     false
   );
 
-  return sc;
+    if (!sc.isDeployed()) {
+        let txnErrors = stringifyTransactionErrors(tx);
+        throw new Error(`Scilla contract was not deployed - status ${sc.status} from ${tx.id}, errors: ${txnErrors}`)
```

Comment:
[It's better](https://hardhat.org/hardhat-runner/docs/advanced/building-plugins#throwing-errors-from-your-plugins) to throw HardhatPluginError. I'm not sure if I followed this suggestion all over the code.

Reply:
The link you posted says:

> If your error originated in your user's code, like a test or script calling one of your functions, you shouldn't use HardhatPluginError.

We only deploy when asked to by the user so I think this applies.

Reply:
That's a good idea! Thanks - I'll do that and merge ..